mayastor-test/e2e/nightly/common/util.go

```diff
@@ -0,0 +1,361 @@
+//	1. The PVC status transitions to bound,
+//	2. The associated PV is created and its status transitions bound
+//	3. The associated MV is created and has a State "healthy"
+
+func MkPVC(volName string, scName string, dynamicClient * dynamic.Interface, kubeInt * kubernetes.Interface) string {
+	fmt.Printf("creating %s, %s\n", volName, scName)
+	// PVC create options
+	createOpts := &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      volName,
+			Namespace: "default",
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			StorageClassName: &scName,
+			AccessModes:      []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: resource.MustParse("64Mi"),
+				},
+			},
+		},
+	}
+
+	// Create the PVC.
+	PVCApi := (*kubeInt).CoreV1().PersistentVolumeClaims
+	_, createErr := PVCApi("default").Create(context.TODO(), createOpts, metav1.CreateOptions{})
+	Expect(createErr).To(BeNil())
+
+	// Confirm the PVC has been created.
+	pvc, getPvcErr := PVCApi("default").Get(context.TODO(), volName, metav1.GetOptions{})
+	Expect(getPvcErr).To(BeNil())
+	Expect(pvc).ToNot(BeNil())
+
+	// Wait for the PVC to be bound.
+	Eventually(func() corev1.PersistentVolumeClaimPhase {
+		return GetPvcClaimStatusPhase(volName, kubeInt)
+	},
+		defTimeoutSecs, // timeout
+		"1s",           // polling interval
+	).Should(Equal(corev1.ClaimBound))
+
+	// Refresh the PVC contents, so that we can get the PV name.
+	pvc, getPvcErr = PVCApi("default").Get(context.TODO(), volName, metav1.GetOptions{})
+	Expect(getPvcErr).To(BeNil())
+	Expect(pvc).ToNot(BeNil())
+
+	// Wait for the PV to be provisioned
+	Eventually(func() *corev1.PersistentVolume {
+		pv, getPvErr := (*kubeInt).CoreV1().PersistentVolumes().Get(context.TODO(), pvc.Spec.VolumeName, metav1.GetOptions{})
+		if getPvErr != nil {
+			return nil
+		}
+		return pv
+
+	},
+		defTimeoutSecs, // timeout
+		"1s",           // polling interval
+	).Should(Not(BeNil()))
+
+	// Wait for the PV to be bound.
+	Eventually(func() corev1.PersistentVolumePhase {
+		return GetPvStatusPhase(pvc.Spec.VolumeName, kubeInt)
+	},
+		defTimeoutSecs, // timeout
+		"1s",           // polling interval
+	).Should(Equal(corev1.VolumeBound))
+
+	msv := GetMSV(string(pvc.ObjectMeta.UID), dynamicClient)
+	Expect(msv).ToNot(BeNil())
+	return string(pvc.ObjectMeta.UID)
+}
+
+// Delete the PVC and verify that
+//	1. The PVC is deleted
+//	2. The associated PV is deleted
+//  3. The associated MV is deleted
+
+func RmPVC(volName string, scName string, dynamicClient * dynamic.Interface, kubeInt * kubernetes.Interface) {
+	fmt.Printf("removing %s, %s\n", volName, scName)
+
+	PVCApi := (*kubeInt).CoreV1().PersistentVolumeClaims
+
+	// Confirm the PVC has been created.
+	pvc, getPvcErr := PVCApi("default").Get(context.TODO(), volName, metav1.GetOptions{})
+	Expect(getPvcErr).To(BeNil())
+	Expect(pvc).ToNot(BeNil())
```

Comment:
Maybe we should move this into its own function as it is used a number of times.

Reply:
Ok, I'll take a look at doing that.

Reply:
Actually, I'm not sure if we gain much at the expense of making the code less clear. I think I'll leave it as-is, at least for now.